File: 5_FL21/Day5/LeetCode_94_Binary_Tree_Inorder_Traversal.cc
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> ans;
        
        if (root != NULL){
            vector<int> tmp = inorderTraversal(root -> left);
            ans.insert(ans.end(), tmp.begin(), tmp.end());
            
            ans.push_back(root -> val);
            
            tmp = inorderTraversal(root -> right);
            ans.insert(ans.end(), tmp.begin(), tmp.end());
        }
        
        return ans;
    }
};

File: 5_FL21/Day5/LeetCode_94_Binary_Tree_Inorder_Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<Integer> ans = new ArrayList<>();
    private void helper(TreeNode root){
        if (root == null) return;
        helper(root.left);
        ans.add(root.val);
        helper(root.right);
    }
    public List<Integer> inorderTraversal(TreeNode root) {
        helper(root);
        return ans;
    }
}